cmd/repo-query/repo-query.go

```diff
@@ -0,0 +1,353 @@
+	if err != nil {
+		return err
+	}
+
+	// not found or ok or error
+	if resp.StatusCode == http.StatusNotFound {
+		fmt.Println("Repo not found❗")
+		fmt.Println("Indexing repo...⏳")
+		err := indexRepo(owner, repo, opts.branch)
+		if err != nil {
+			return err
+		}
+
+		for {
+			fmt.Printf("\nWant to ask a question about %s/%s?\n", owner, repo)
+			fmt.Printf("> ")
+			// read input
+			scanner := bufio.NewScanner(os.Stdin)
+			if scanner.Scan() {
+				input := scanner.Text()
+				err := askQuestion(input, owner, repo, opts.branch)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	} else if resp.StatusCode == http.StatusOK {
+		fmt.Println("Repo found ✅")
+
+		// this should be an infinite loop
+		for {
+			fmt.Printf("\nWant to ask a question about %s/%s?\n", owner, repo)
+			fmt.Printf("> ")
+			// read input
+			scanner := bufio.NewScanner(os.Stdin)
+			if scanner.Scan() {
+				input := scanner.Text()
+				err := askQuestion(input, owner, repo, opts.branch)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	} else {
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return err
+		}
+
+		fmt.Printf("An error occurred: %v\n", string(body))
+		return errors.New("error while checking if repo is indexed")
+	}
+}
+
+type indexPostRequest struct {
+	Owner  string `json:"owner"`
+	Name   string `json:"name"`
+	Branch string `json:"branch"`
+}
+
+func indexRepo(owner string, repo string, branch string) error {
+	indexPostReq := &indexPostRequest{
+		Owner:  owner,
+		Name:   repo,
+		Branch: branch,
+	}
+
+	indexPostJSON, err := json.Marshal(indexPostReq)
+	if err != nil {
+		return err
+	}
+
+	responseBody := bytes.NewBuffer(indexPostJSON)
+	resp, err := http.Post(fmt.Sprintf("%s/embed", repoQueryURL), "application/json", responseBody)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return err
+		}
+
+		fmt.Printf("An error occurred: %v\n", string(body))
+		return errors.New("error while indexing repository")
+	}
+
+	// listen for SSEs and send data,event pairs to processIndexChunk
+	reader := bufio.NewReader(resp.Body)
+	for {
+		line, err := reader.ReadString('\n')
+		// if err is eof then break
+		if err == io.EOF {
+			return nil
+		} else if err != nil {
+			return err
+		}
+
+		if line == "\n" {
+			continue
+		}
+
+		if strings.HasPrefix(line, "event: ") {
+			chunk := line
+			for {
+				line, err := reader.ReadString('\n')
+				if err != nil {
+					return err
+				}
+
+				if line == "\n" {
+					break
+				}
+
+				chunk += line
+			}
+
+			err := processIndexChunk(chunk)
+			if err != nil {
+				return err
+			}
+		}
+	}
+}
+
+func processIndexChunk(chunk string) error {
+	chunkLines := strings.Split(chunk, "\n")
+	eventLine := chunkLines[0]
+	event := strings.Split(eventLine, ": ")[1]
```

Comment:
This is also rather brittle but probably just needs some inline documentation on what's going on.